src/celengine/observer.cpp

```diff
@@ -816,25 +814,19 @@ void Observer::setFrame(ObserverFrame::CoordinateSystem cs, const Selection& ref
 /*! Set the observer's reference frame. The position of the observer in
  *  universal coordinates will not change.
  */
-void Observer::setFrame(const ObserverFrame& f)
+void Observer::setFrame(const ObserverFrame::SharedConstPtr& f)
 {
-    if (frame != &f)
+    if (frame)
     {
-        auto* newFrame = new ObserverFrame(f);
-
-        if (frame != nullptr)
-        {
-            convertFrameCoordinates(newFrame);
-            delete frame;
-        }
-        frame = newFrame;
+        convertFrameCoordinates(f);
```

Comment:
could you check `convertFrameCoordinates`? i'm not sure that your change is equivalent to the original code